ext/random/randomizer.c

```diff
@@ -291,17 +327,28 @@ PHP_METHOD(Random_Randomizer, getBytes)
 
 	size_t length = (size_t)user_length;
 	retval = zend_string_alloc(length, 0);
+	char *rptr = ZSTR_VAL(retval);
 
-	while (total_size < length) {
+	size_t to_read = length;
+	while (to_read > 0) {
 		php_random_result result = engine.algo->generate(engine.state);
 		if (EG(exception)) {
 			zend_string_free(retval);
 			RETURN_THROWS();
 		}
-		for (size_t i = 0; i < result.size; i++) {
-			ZSTR_VAL(retval)[total_size++] = (result.result >> (i * 8)) & 0xff;
-			if (total_size >= length) {
-				break;
+		if (EXPECTED(result.size == sizeof(uint64_t))) {
+			rptr = bulk_convert_generated_result_to_char_64(rptr, result.result, &to_read);
+		} else if (EXPECTED(result.size == sizeof(uint32_t))){
+			rptr = bulk_convert_generated_result_to_char_32(rptr, result.result, &to_read);
+		} else {
```

Comment:
Is there a possibility that this `else` will be executed?

Reply:
Yes. It can happen with userland engines that return single bytes:

```php
final class MyEngine implements \Random\Engine {
  public function generate(): string { return 'x'; }
}

$r = new \Random\Randomizer(new MyEngine());

var_dump($r->getBytes(10));
```